sklearn/datasets/tests/test_base.py

```diff
@@ -129,6 +128,20 @@ def test_load_files_wo_load_content(
     assert res.get("data") is None
 
 
+def test_load_files_allowed_extensions(tmp_path):
+    """Check the behaviour of `allowed_extension` in `load_files`."""
+    d = tmp_path / "sub"
+    d.mkdir()
+    p1 = d / "file1.txt"
+    p1.touch()
+    p2 = d / "file2.json"
+    p2.touch()
+    p3 = d / "file3.json"
+    p3.touch()
+    res = load_files(tmp_path, allowed_extensions=[".txt"])
+    assert set([str(p1)]) == set(res.filenames)
```

Comment:
We can add another file with a new extension `d / "file4.md"` and check `load_files(..., allowed_extensions=[".json", ".md"]` gives the expected results.

This way we check for the multiple file extension case.